src/sql/azdata.proposed.d.ts

```diff
@@ -432,6 +432,54 @@ declare module 'azdata' {
 		azurePortalEndpoint?: string;
 	}
 
+
+	export namespace diagnostics {
+		/**
+		 * Represents a diagnostics provider of accounts.
+		 */
+		export interface ErrorDiagnosticsProviderMetadata {
+			/**
+			 * The name of the provider (ex. a connection provider) that a diagnostics provider will handle errors for.
+			 * Note: only ONE diagnostic provider per id/name at a time.
+			 */
+			id: string;
```

Comment:
nit: Would still suggest renaming this to something like `targetProviderId` to make it clear this isn't supposed to be an ID for this provider but rather the ID of another provider.